subprojects/junit-jupiter/src/test/java/org/mockitousage/StrictnessTest.java

```diff
@@ -53,6 +53,28 @@ void session_checks_for_strict_stubs() {
         assertThat(result.getThrowable().get()).isInstanceOf(UnnecessaryStubbingException.class);
     }
 
+    @MockitoSettings(strictness = Strictness.STRICT_STUBS)
+    static class StrictStubsNotReportedOnTestFailure {
+        @Mock
+        private Function<Integer, String> rootMock;
+
+        @Test
+        void should_not_throw_exception_on_strict_stubs_because_of_test_failure() {
+            Mockito.when(rootMock.apply(10)).thenReturn("Foo");
+            fail("Test failed");
+        }
+    }
+
+    @Test
+    void session_does_not_check_for_strict_stubs_on_test_failure() {
+        TestExecutionResult result = invokeTestClassAndRetrieveMethodResult(StrictStubsNotReportedOnTestFailure.class);
+
+        assertThat(result.getStatus()).isEqualTo(TestExecutionResult.Status.FAILED);
+        Throwable throwable = result.getThrowable().get();
+        assertThat(throwable).isInstanceOf(AssertionError.class);
```

Comment:
Can you make the assertion stronger by using a more specific exception? You can create a custom exception type in the test.

Reply:
Ok, I see you already did it.

Reply:
Yes ;)